core/src/main/java/hudson/FilePath.java

```diff
@@ -563,40 +597,53 @@ public Void invoke(File dir, VirtualChannel channel) throws IOException, Interru
      * @param compression
      *      Compression mode of this tar file.
      * @since 1.292
-     * @see #untarFrom(InputStream, TarCompression)
+     * @see #untar2(FilePath, TarCompression)
+     * @deprecated
      */
     public void untar(final FilePath target, final TarCompression compression) throws IOException, InterruptedException {
+        untar2(target, compression);
```

Comment:
I'm not sure this is the best idea.
I am sure someone un archives something with lots of files and this will cause GC churn.
are we aware of callers outside `unstash` have we looked at how this could be impacting them?

Reply:
@joseblas did you have a chance to review this comment?

Reply:
Sorry, I forgot to close this. Is a fair point from James.